packages/web-components/src/theme/set-theme.ts

```diff
@@ -10,12 +9,12 @@ const tokenNames = Object.keys(tokens) as (keyof Theme)[];
  */
 export const setTheme = (theme: Theme) => {
   for (const t of tokenNames) {
-    tokens[t].withDefault(theme[t] as string);
+    document.body.style.setProperty(tokens[t] as string, theme[t] as string);
```

Comment:
This is not at all pressing for this PR but in Fluent React we've banned direct global access like this as it has lead to several subtle, difficult to debug issues. It is also a problem with SSR as `document` will not exist in that case.

In React we can handle this with context but for a function like this we've been adding a, typically optional, argument to the function that allows the caller to provide the `document/window/whatever-other-global` the function requires.

[Example](https://github.com/microsoft/fluentui/pull/29962/files#diff-753cbfd679dfe0f09b99a327bb5e000ea0a3699d03b75c0e3879690b0e1fd420)

Reply:
We'll have an SSR story here for things like this. 

We should look to address this, but this essentially does what _was_ being done with more code via DesignToken.

Reply:
Works for me, just wanted you to be aware :)

Reply:
For sure - we’ll need to solve a couple places.